### PR TITLE
Clean Pushgateway metrics in CronJob

### PR DESCRIPTION
**Pull request by suiguoxin**

### CODE REVIEW

File: src/prometheus-pushgateway/build/prometheus-pushgateway.common.dockerfile [hunk 1]
```diff
@@ -1,18 +1,4 @@
...
-# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
-# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
-# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
-# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
```
Comment (Binyang2014) [inline src/prometheus-pushgateway/build/prometheus-pushgateway.common.dockerfile:2]: Please make sure this license header is valid

File: src/prometheus-pushgateway/config/prometheus-pushgateway.yaml [hunk 2]
```diff
@@ -1,21 +1,10 @@
...
+# Licensed under the MIT License.
 
 service_type: "common"
 
 port: 9097
-scrape_interval: 30
+scrape_interval: 30 # seconds
+metrics-cleaner:
+  schedule: '0 0 * * *'
+  clean-period: 7 # days
```
Comment (Binyang2014) [inline src/prometheus-pushgateway/config/prometheus-pushgateway.yaml:10]: For staled metrics. If it doesn't be deleted form push gateway, will it be reported to prometheus? I mean if a job finished at 10:00 am, we run clean daemon at 10:00 pm. Will we still see the job metrics between 10:00 am to 10:00 pm

File: src/prometheus-pushgateway/build/prometheus-pushgateway.common.dockerfile (hunk 1 above)
Comment (suiguoxin) [reply src/prometheus-pushgateway/build/prometheus-pushgateway.common.dockerfile:2]: Yes, we use this simplified version now. It is not applied across the repo yet.

File: src/prometheus-pushgateway/config/prometheus-pushgateway.yaml (hunk 2 above)
Comment (suiguoxin) [reply src/prometheus-pushgateway/config/prometheus-pushgateway.yaml:10]: > For staled metrics. If it doesn't be deleted form push gateway, will it be reported to prometheus? I mean if a job finished at 10:00 am, we run clean daemon at 10:00 pm. Will we still see the job metrics between 10:00 am to 10:00 pm

Yes

**Comment by suiguoxin** [comment]
Cronjob is not appropriate for this case. Use #5553 instead

**Pull request closed without merging**